qcodes/instrument_drivers/agilent/E8527D.py

```diff
@@ -1,3 +1,4 @@
+import warnings
 from typing import Any
```

Comment:
you will now also need to import Optional from typing

Reply:
why? could you please explain me? I already saw this on the mypy test. Does it has to do that the "frequency_option" could be either a boolean or a str depending on the if statement?

Reply:
ahh, I guess it is for "step_attenuator: Optional[bool] "

Reply:
No, this is because we changed the type of `step_attenuator` so it now can be either None or True or False by using Optional on line 19. 

I suggested this such that we change the default value for `step_attenuator` to None such that if any user explicitly sets it to any of the previously allowed values (True or False) they will trigger the warning that it is unused and they should stop setting it. 

We are trying to type almost all the code within QCoDeS as this helps catch many types of errors in the code. If you are interested in learning more I recommend reading the introduction to the MyPy docs here https://mypy.readthedocs.io/en/stable/getting_started.html#getting-started 

Hope that makes sense :)

Reply:
> ahh, I guess it is for "step_attenuator: Optional[bool] "
Yes I'm sorry I did not see this before I wrote my reply

Reply:
thanks!